src/mame/sinclair/sprinter.cpp

```diff
@@ -1506,10 +1521,20 @@ void sprinter_state::do_cpu_wait(bool is_io)
 
 TIMER_CALLBACK_MEMBER(sprinter_state::irq_on)
 {
-	m_maincpu->pulse_input_line(INPUT_LINE_IRQ0, attotime::from_ticks(26, m_maincpu->clock()));
+	if (!m_hold_irq)
+	{
+		m_maincpu->set_input_line(INPUT_LINE_IRQ0, ASSERT_LINE);
+		m_irq_off_timer->adjust(attotime::from_ticks(32, m_maincpu->unscaled_clock()));
+	}
 	update_int(false);
 }
 
+TIMER_CALLBACK_MEMBER(sprinter_state::irq_off)
+{
+	m_hold_irq = 0;
+	m_maincpu->set_input_line(INPUT_LINE_IRQ0, CLEAR_LINE);
+}
```

Comment:
This can be called two ways now – from `m_irq_off_timer` expiring, and from an interrupt acknowledge cycle on the main CPU.  If it’s called from an interrupt acknowledge cycle, should it cancel the timer?  If it doesn’t, this sequence of events can happen:

* `irq_on` asserts interrupt and starts `m_irq_off_timer`
* Interrupt acknowledge cycle causes `irq_off` to be called, clearing interrupt
* `m_irq_off_timer` fires, causing `irq_off` to be called again

It’s probably harmless given what it’s doing, but it will go through another timer update and then go through the scheduler again to clear the IRQ line.